File: Javascript/Basic Recursion/Problme3.js

// print  N to 1 

function printNums(n){
    if(n==0){
      return 
    }
    console.log(n);
    printNums(n-1);
  }
  
  
  function main(){
    let n=5;
    printNums(n)
  }
  
  
  main()
  

File: C++/Basic Recursion/Problme3.cpp


#include <iostream>
using namespace std;


void printNums(int i,int n){
  if(i>n){
    return;
  }
  std::cout << n << std::endl;
  
  printNums(i,n-i);
}

int main() 
{
    int n=10;
    printNums(1,n);
    return 0;
}